app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt

```diff
@@ -82,6 +84,51 @@ class PreviewBadge : View {
                 View.MeasureSpec.makeMeasureSpec(calculatedHeight, View.MeasureSpec.EXACTLY))
     }
 
+    public override fun onSaveInstanceState(): Parcelable? {
+        val bundle = Bundle()
+        bundle.putParcelable("superState", super.onSaveInstanceState())
+        bundle.putBoolean("ifFlash", this.ifFlash)
+        bundle.putBoolean("ifMarquee", this.ifMarquee)
+        bundle.putInt("badgeSpeed", this.badgeSpeed)
+        bundle.putInt("badgeMode", when (badgeMode) {
```

Comment:
put string and use enum value of to deserialize

Reply:
Let me use ordinal, it would be better and more memory efficient